src/lib/eventManager.js

```diff
@@ -1,5 +1,61 @@
-export function setupEventListeners(root) {}
+const eventManager = new Map();
 
-export function addEvent(element, eventType, handler) {}
+function createSyntheticEvent(event) {
+  let propagationStopped = false;
+  return {
+    type: event.type,
+    target: event.target,
+    currentTarget: event.target,
+    preventDefault() {
+      event.preventDefault();
+    },
+    stopPropagation() {
+      propagationStopped = true;
+      event.stopPropagation();
+    },
+    isPropagationStopped() {
+      return propagationStopped;
+    },
+    nativeEvent: event,
+  };
+}
 
-export function removeEvent(element, eventType, handler) {}
+function handleGlobalEvent(event) {
+  event = createSyntheticEvent(event);
+  const eventType = event.type;
+  const handlers = eventManager.get(eventType);
+
+  let currentElement = event.target;
+  while (currentElement && !event.isPropagationStopped()) {
+    if (handlers.has(currentElement)) {
+      const handler = handlers.get(currentElement);
+      handler(event);
+    }
+    currentElement = currentElement.parentElement;
```

Comment:
핸들러 함수가 존재하지 않거나 핸들러 함수 실행에 실패했을 때? 에 대한 예외처리도 있으면 좋을 것 같아요~!

```suggestion
    if (handlers.has(currentElement)) {
      const handler = handlers.get(currentElement);
      try {
        handler(event);
      } catch (error) {
        // 커스텀한 에러 객체 생성도 좋은 방법인 것 같습니다
      }
    }
    currentElement = currentElement.parentElement;
```

Reply:
handler의 유무는 위에서 체크하고 있어 괜찮을것 같아보이긴 한데 혹시 함수실행에 실패했을때에는 어떠한 경우가 있을까요?

Reply:
handlers에 has로 currentElement 유무를 확인한 뒤 get을 해주어 handler를 가져오다보니 내부적으로 유효하지 않은 값이 할당되어있을 가능성을 고려해봤는데 잘못 생각한 것 같네요..ㅎ 다만 handlers.has → GC 수집 → handlers.get 이런 흐름도 예측해볼 수 있긴한데 너무 이론적인 엣지 케이스인 것 같아 그냥 무시하셔도 될 듯 합니다 ㅋㅋ..

함수 실행 실패는 handler 수행 동작 과정에서 어떤 오류가 발생해 실패한 경우 프로그램에 지장을 주지 않게 하기 위해 try catch로 적절하게 예외처리하면 좋지 않을까 하는 생각이었어요! 저도 구체적으로 떠오르진 않아 GPT한테 물어봤는데,
- 함수 파라미터 문제
- 함수 내부에서 참조하는 외부 값이 변경 또는 삭제
- this 컨텍스트 바인딩 문제
- 이밴트 객체 속성 접근 문제
뭐 이런 것들이 있을 수 있다고는 합니다..!

Reply:
역시 도운님 ,,, 예시 최고 bb